transport-native-kqueue/src/main/java/io/netty/channel/kqueue/BsdSocket.java

```diff
@@ -80,6 +85,70 @@ long sendFile(DefaultFileRegion src, long baseOffset, long offset, long length)
         return ioResult("sendfile", (int) res);
     }
 
+    int connectx(InetSocketAddress source, InetSocketAddress destination, IovArray data, boolean tcpFastOpen)
+            throws IOException {
+        int flags = tcpFastOpen ? connectResumeOnReadWrite() | connectDataIdempotent() : 0;
+        return connectx(source, destination, data, flags);
+    }
+
+    int connectx(InetSocketAddress source, InetSocketAddress destination, IovArray data, int flags) throws IOException {
+        int sourceInterface = 0;
```

Comment:
if the interface never needs to be specified, consider inlining it to 0 in the function call. also plz add docs clarifying the choice of 0 (I couldn't find links to 1st party docs but some 3rd party ones below.

https://www.manpagez.com/man/2/connectx/
https://www.manpagez.com/man/3/if_nametoindex/